javascript/packages/cli/src/actions/setup.ts

```diff
@@ -0,0 +1,235 @@
+        new Promise<void>(async (resolve, reject) => {
+          let result = options[binary];
+          if (!result) {
+            console.log("options", options, "binary", binary);
+            throw new Error("Binary is not defined");
+          }
+          const { url, name } = result;
+
+          if (!url) throw new Error("No url for downloading, was provided");
+
+          const response = await fetch(url);
+
+          if (!response.ok)
+            throw Error(response.status + " " + response.statusText);
+
+          const contentLength = response.headers.get(
+            "content-length",
+          ) as string;
+          let loaded = 0;
+
+          const progressBar = multibar.create(parseInt(contentLength, 10), 0);
+          const reader = response.body?.getReader();
+          const writer = fs.createWriteStream(path.resolve(name));
+
+          while (true) {
+            const read = await reader?.read()!;
+            if (read?.done) {
+              writer.close();
+              resolve();
+              break;
+            }
+
+            loaded += read.value.length;
+            progressBar.increment();
+            progressBar.update(loaded, {
+              filename: name,
+            });
+            writer.write(read.value);
+          }
+        }),
+      );
+    }
+
+    await Promise.all(promises);
+    multibar.stop();
+    console.log(
+      decorators.cyan(
+        `\n\nPlease add the current dir to your $PATH by running the command:\n`,
+      ),
+      decorators.blue(`export PATH=${process.cwd()}:$PATH\n\n`),
+    );
+  } catch (err) {
+    console.log(
+      `\n ${decorators.red("Unexpected error: ")} \t ${decorators.bright(
+        err,
+      )}\n`,
+    );
+  }
+};
+
+// Retrieve the latest release for polkadot
+const latestPolkadotReleaseURL = async (
+  repo: string,
+  name: string,
+): Promise<[string, string]> => {
+  try {
+    const releases = await fetch(
+      `https://api.github.com/repos/paritytech/${repo}/releases`,
+    );
+
+    let obj: any;
+    let tag_name;
+
+    const allReleases = await releases.json();
+    const release = allReleases.find((r: any) => {
+      obj = r?.assets?.find((a: any) => a.name === name);
+      return Boolean(obj);
+    });
+
+    tag_name = release.tag_name;
+
+    if (!tag_name) {
+      throw new Error(
+        "Should never come to this point. Tag_name should never be undefined!",
+      );
+    }
+
+    return [
+      `https://github.com/paritytech/${repo}/releases/download/${tag_name}/${name}`,
+      convertBytes(obj.size),
+    ];
+  } catch (err: any) {
+    if (err.code === "ENOTFOUND") {
+      throw new Error("Network error.");
+    } else if (err.response && err.response.status === 404) {
+      throw new Error(
+        "Could not find a release. Error 404 (not found) detected",
+      );
+    }
+    throw new Error(
+      `Error status: ${err?.response?.status}. Error message: ${err?.response}`,
+    );
+  }
+};
```

Comment:
That's why I don't like axios and prefer [wretch](https://github.com/elbywan/wretch), the code can be simplified like this:

```ts
wretch("anything")
  .get()
  .notFound(error => { /* ... */ })
  .unauthorized(error => { /* ... */ })
  .error(418, error => { /* ... */ })
  .res(response => /* ... */)
  .catch(error => { /* uncaught errors */ })
  ```

Reply:
Yes, we are moving away from `axios`. I think @wirednkod is doing this in another pr.